pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -96,12 +103,19 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 
 	applog.Info("build template")
 	// build template to applicationconfig & component
-	ac, comps, err := appParser.GenerateApplicationConfiguration(appfile, app.Namespace)
+	ac, comps, err := appParser.GenerateApplicationConfiguration(generatedAppfile, app.Namespace)
 	if err != nil {
 		applog.Error(err, "[Handle GenerateApplicationConfiguration]")
 		app.Status.SetConditions(errorCondition("Built", err))
 		return handler.handleErr(err)
 	}
+	handler.specDiff, handler.newRevision, err = r.GenerateRevision(ctx, generatedAppfile, app, ac, comps)
```

Comment:
Should add a if-else here:
- If it is rollout-template mode, just create the revision, no appConfig
- otherwise, create appConfig

Right?

Reply:
yes but that logic is in apply.go